lib/modules/versioning/same-major/index.spec.ts

```diff
@@ -0,0 +1,70 @@
+import sameMajor from '.';
+
+describe('modules/versioning/same-major/index', () => {
+  describe('.isGreaterThan(version, other)', () => {
+    it('should return true', () => {
+      expect(sameMajor.isGreaterThan('4.0.0', '3.0.0')).toBeTrue(); // greater
+    });
+
+    it('should return false', () => {
+      expect(sameMajor.isGreaterThan('2.0.2', '3.1.0')).toBeFalse(); // less
+      expect(sameMajor.isGreaterThan('3.1.0', '3.0.0')).toBeFalse(); // same major -> equal
+      expect(sameMajor.isGreaterThan('3.0.0', '3.0.0')).toBeFalse(); // equal
+      expect(sameMajor.isGreaterThan('a', '3.0.0')).toBeFalse(); // invalid versions
+    });
+  });
+
+  describe('.matches(version, range)', () => {
+    it('should return true when version has same major', () => {
+      expect(sameMajor.matches('1.0.1', '1.0.0')).toBeTrue();
+      expect(sameMajor.matches('1.0.1', '^1.0.0')).toBeTrue(); // coverage for when ranges are passed to the function
```

Comment:
Why are we supporting range syntax like this?

Reply:
We return the version as is without massaging if it's a range, which leads to this behaviour. Should I throw an error instead or log a warning instead reminding the user to not use same-major with ranges.

Reply:
I'm ok with "accidentally" supporting more than we want (e.g. because semver-coerced supports it) but we shouldn't test or document it as that makes it semi-official